src/item.cpp

```diff
@@ -3871,6 +3871,31 @@ long item::ammo_required() const {
     return res;
 }
 
+bool item::ammo_consume( int qty ) {
+    if ( qty > 0 || qty <= ammo_remaining() ) {
```

Comment:
Code quality:
The project coding style is `if( x )` - no space after `if`, space after the parenthesis.
You could invert that `if` - it contains pretty much the entire body of the function. As in:
```
if( qty <= 0 || qty > ammo_remaining() ) {
    return false;
}

// Body of what you have in the 'if' right now
```

That would save on indentations.

Quitting on `qty > ammo_remaining()` sounds like a bug risk. You should probably print a debugmsg in such a case. Otherwise it may lead to cases where ammo is not consumed at all.